lib/std/child_process.zig

```diff
@@ -126,8 +126,10 @@ pub const ChildProcess = struct {
     pub fn spawn(self: *ChildProcess) SpawnError!void {
         if (builtin.os.tag == .windows) {
             return self.spawnWindows();
-        } else {
+        } else if (std.process.can_spawn) {
             return self.spawnPosix();
+        } else {
+            unreachable;
```

Comment:
I think this spawn function should be untouched, or if you want to make the compile error clearer, put at the top of it:

```zig
if (!std.process.can_spawn) {
    @compileError("the target operating system cannot spawn processes");
}
```